aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py

```diff
@@ -77,23 +77,26 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
                 "dependencies directory was not provided and downloading dependencies is disabled."
             )
 
-        self.actions = [
-            CopySourceAction(source_dir=self.source_dir, dest_dir=self.scratch_dir, excludes=self.EXCLUDED_FILES)
-        ]
+        # if we're building in the source directory, we don't have to copy the source code
+        self.actions = (
+            []
+            if self.build_dir == self.source_dir
+            else [CopySourceAction(source_dir=self.source_dir, dest_dir=self.build_dir, excludes=self.EXCLUDED_FILES)]
+        )
 
         if self.download_dependencies:
             self.actions.append(
                 NodejsNpmWorkflow.get_install_action(
                     source_dir=source_dir,
-                    artifacts_dir=self.scratch_dir,
+                    install_dir=self.build_dir,
```

Comment:
I am not sure if I miss something or not, but I am not able to find the build_dir change in this class

Reply:
never mind, I found it in the `BaseWorkflow` class.